Makefile

```diff
@@ -77,7 +77,7 @@ unit-test-db:
 	docker-compose -f docker-compose.test.yml up -d db-unit-test
 	
 	# Wait for the database to be ready
-	sleep 5
+	docker-compose -f docker-compose.wait-for-it.yml run --rm wait wait-for-it -h db-unit-test -p 5432 -t 60
```

Comment:
Could we forgo docker-compose and use netcat instead?

Somewhere in the Makefile (maybe at the end of the file), you can have:
```
define waitforit
	@num=0; while ! nc -z localhost $(1); do \
		sleep 1; \
		echo $$num; \
		((num = num + 1)); \
		if [ $$num -eq $(2) ]; then\
		echo "Timed out..."; \
		break; \
		fi \
	done
endef
```
And then here you can have
```
$(call waitforit, 5432, 60)
```

Reply:
Makes sense to me! I was thinking about using docker-compose to be able to use the service name (db-unit-test), but since we expose a local port on all of the containers, that step is not necssary.

Reply:
After trying it - it looks like using the external port is successful but our attempt to run the migrate command fails due to connection refused. I'm think we should continue using this strategy.